sale_loyalty_limit/models/sale_order.py

```diff
@@ -0,0 +1,132 @@
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _program_check_compute_points(self, programs):
+        res = super()._program_check_compute_points(programs)
+        programs_with_additional_errors = {}
+        # Iterate through the programs that initially have no errors
+        for program, result in res.items():
+            if "error" not in result:
+                for rule in program.rule_ids:
```

Comment:
Take the logic in a separate method and check here if all the rules are limited. Only in that case we flag the program in the dictionary

Reply:
This generates me a bit of doubt, let's say we have 2 rules, we would have two cases:
1. one of the rules has "rule_max_customer_application" set and the other one doesn't, in that case no error should be generated and the promotion is applied. (In this case the functionality of another module to enforce all the rules would be the one that would generate the error)
2. Both rules have "rule_max_customer_application" set and in this case an error must be generated to not allow the promotion to be applied (in this case is where I am looking for confirmation). If one rule has the limit set to 1 and another rule has the limit set to 2, does the lower limit prevail?